pilot/pkg/model/push_context.go

```diff
@@ -562,6 +567,11 @@ func (ps *PushContext) InitContext(env *Environment) error {
 	ps.Env = env
 	var err error
 
+	// Must be initialized first
+	// as initServiceRegistry/VirtualServices/Destrules
+	// use the default export map
+	ps.initDefaultExportMaps()
```

Comment:
It is a waste of resource

Reply:
But the data structures are stored in push context. Right now, its not that heavy because we allow only one entry in each exportTo. So, its effectively scanning an array of size 1

Reply:
And we'll need to reload mesh config ( I think we have some code already ?)

Reply:
yes thats already there. @ymesika added it long time ago. And during reload, it triggers configUpdate event